src/yaml.cr

```diff
@@ -147,4 +148,28 @@ module YAML
   def self.dump(object, io : IO)
     object.to_yaml(io)
   end
+
+  # Determines if the scaler is reserved
+  def self.reserved_scalar(value : String)
```

Comment:
How can we get back on track here. I don't agree with some of @ysbaddaden's suggestions. Can we get someone like @asterite to weigh in here? The method of matching with regex comes closest to matching the YAML spec.